lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1045,6 +1038,19 @@ static LogicalResult verifyInstanceOp(InstanceOp instance) {
     }
   }
 
+  return success();
+}
+
+/// Verify the correctness of an InstanceOp.
+static LogicalResult verifyInstanceOp(InstanceOp instance) {
+
+  // Check that this instance is inside a module.
+  auto module = instance->getParentOfType<FModuleOp>();
+  if (!module) {
+    instance.emitOpError("should be embedded in a 'firrtl.module'");
+    return failure();
+  }
```

Comment:
I stumbled across this some time ago and was wondering if this couldn't be moved into the `*.td` file as `HasParent` trait on the op?

Reply:
`HasParent` [only looks one level up](https://github.com/llvm/llvm-project/blob/387f95541bdc575f75bba95d323b51198d3b9e2b/mlir/include/mlir/IR/OpDefinition.h#L1215), while we need to be able to put these in `firrtl.when`s regions. Maybe we should add a new optrait?

Reply:
Ah good point!